package/src/integration.ts

```diff
@@ -12,48 +27,135 @@ export default defineIntegration({
   optionsSchema,
   plugins: [...corePlugins],
   setup({ options }) {
-	const { resolve } = createResolver(import.meta.url)
+	// Create resolve helper
+	const { resolve } = createResolver(import.meta.url);
+
+	// Check if verbose logging is enabled
+	const isVerbose = options.verbose;
+	
+	// Create Gist Logger interface
+	const gistLogger = async (
+		logger: AstroIntegrationLogger, 
+		type: "info"|"warn"|"error", 
+		message: string,
+		checkVerbose: boolean,
+		) => {
+			// if checkVerbose is true and isVerbose is true, log the message
+			if (checkVerbose && isVerbose) {
+				if (type === "info") {
+					logger.info(message);
+				} else if (type === "warn") {
+					logger.warn(message);
+				} else if (type === "error") {
+					logger.error(message);
+				} 
+			}
+			// if checkVerbose is false, force log the message
+			if (!checkVerbose) {
+				if (type === "info") {
+					logger.info(message);
+				} else if (type === "warn") {
+					logger.warn(message);
+				} else if (type === "error") {
+					logger.error(message);
+				} 
+		} };
 
 	return {
 	  "astro:config:setup": ({ 
 		watchIntegration, addVirtualImports, logger, addDts
 	}) => {
-		logger.info("Setting up Astro Gists Integration.")
-		const configSetup = logger.fork("astro-gists/config:setup")
-		
+
+		// Create a logger for the setup events
+		const configLogger = logger.fork("astro-gists : setup");
+
+		// Create a verbose logger
+		const verboseLogger = (
+			type: "info"|"warn"|"error", 
+			message:string
+			) => gistLogger(configLogger, type, message, true);
+
+		// Create a logger that ignores the verbose check
+		const ignoreVerboseLoggerCheck = (
+			type: "info"|"warn"|"error",
+			message:string
+			) => gistLogger(configLogger, type, message, false);
```

Comment:
Not 100% sold on these either, feel like you could just have one (because it is still nice it abstracts passing the `configLogger`) and you could just pass a `force` boolean when you use it which just passes that to the `gistLogger`